commit 2f8e73b64902de7757dccc7e416477b5e72f04bd
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Mon Mar 20 15:47:57 2023 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Mon Mar 20 12:47:57 2023 -0700

    Ha authentication fix (#1535)
    
    * Added the missing authentication functions for server job process.
    
    * notify the server state change to the running jobs.
    
    * codestyle fix.
    
    * renamed a logger.
    
    ---------
    
    Co-authored-by: Chester Chen <512707+chesterxgchen@users.noreply.github.com>
    Co-authored-by: Yuan-Ting Hsieh (謝沅廷) <yuantingh@nvidia.com>
---
 nvflare/private/fed/server/server_engine.py | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 73f373c2..ea889084 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -56,6 +56,7 @@ from nvflare.fuel.utils.zip_utils import zip_directory_to_bytes
 from nvflare.private.admin_defs import Message, MsgHeader
 from nvflare.private.defs import CellChannel, CellMessageHeaderKeys, RequestHeader, TrainingTopic, new_cell_message
 from nvflare.private.fed.server.server_json_config import ServerJsonConfigurator
+from nvflare.private.fed.server.server_state import ServerState
 from nvflare.private.fed.utils.fed_utils import security_close
 from nvflare.private.scheduler_constants import ShareableHeader
 from nvflare.security.logging import secure_format_exception
@@ -196,6 +197,7 @@ class ServerEngine(ServerEngineInternalSpec):
                 job_clients,
                 snapshot,
                 self.server.cell,
+                self.server.server_state,
             )
 
             self.engine_info.status = MachineStatus.STARTED
@@ -225,7 +227,17 @@ class ServerEngine(ServerEngineInternalSpec):
         self.engine_info.status = MachineStatus.STOPPED
 
     def _start_runner_process(
-        self, args, app_root, run_number, app_custom_folder, open_ports, job_id, job_clients, snapshot, cell: Cell
+        self,
+        args,
+        app_root,
+        run_number,
+        app_custom_folder,
+        open_ports,
+        job_id,
+        job_clients,
+        snapshot,
+        cell: Cell,
+        server_state: ServerState,
     ):
         new_env = os.environ.copy()
         if app_custom_folder != "":
@@ -252,6 +264,12 @@ class ServerEngine(ServerEngineInternalSpec):
             + str(cell.get_internal_listener_url())
             + " -u "
             + str(cell.get_root_url_for_child())
+            + " --host "
+            + str(server_state.host)
+            + " --port "
+            + str(server_state.service_port)
+            + " --ssid "
+            + str(server_state.ssid)
             + " --set"
             + command_options
             + " print_conf=True restore_snapshot="
